akka-http-core/src/main/resources/reference.conf

```diff
@@ -260,6 +260,21 @@ akka.http {
       # be increased for high bandwidth-delay-product connections.
       incoming-stream-level-buffer-size = 512kB
 
+      # For incoming requests, the infrastructure collects at least the given number of bytes before dispatching a HttpRequest.
+      # If all request data is received before or when the threshold is reached, the entity data is dispatched as a strict entity
+      # which allows more efficient processing of the request data without involving streams.
+      #
+      # If the given value is 0, the request is immediately dispatched and always carries a stream of data.
+      # You can use value `1` to create a strict entity if the request contains at most a single data frame, and a streamed
+      # entity otherwise.
+      #
+      # If a strict entity is dispatched, the data size will be at least the amount given here.
+      #
+      # Note, that if enabled, requests that expect data (endStream = false on HEADERS) but never received a DATA frame will never be
```

Comment:
@raboof WDYT about this? I think this is no constraint at all for gRPC because every request needs to have a payload AFAIU.

But even for usual HTTP traffic this seems hardly a concern. A pattern that doesn't work when enabled would be a kind of long-polling where the client is still allowed to send some control messages but doesn't do so at the start of the request.

A DoS concern could be that an attacker never sends over an expected request entities. In that case, the behavior with the feature enabled would be better than before since we would allocate even less resources than with the feature disabled.

In any case, it would make sense to guard against these kinds of attacks with more timeouts, e.g. `max-time-to-first-entity-byte`.

Reply:
> I think this is no constraint at all for gRPC because every request needs to have a payload AFAIU.

I think so too. That payload might be a `google.protobuf.Empty` which serializes to an empty bytearray, but it'd still be wrapped in 2 bytes of gRPC framing, so there would be a DATA frame after all.

Reply:
we can't make this enabled by default?

Reply:
Yes, it's because of this note that I wouldn't want to enable it by default right now. We should enable it in akka-grpc, though.